addons/sale/static/src/js/product_card/product_card.js

```diff
@@ -0,0 +1,25 @@
+import { Component } from "@odoo/owl";
+import { BadgeExtraPrice } from "../badge_extra_price/badge_extra_price";
+import { ProductProduct } from "../models/product_product";
+import { ProductTemplateAttributeLine } from "../models/product_template_attribute_line";
+
+export class ProductCard extends Component {
+    static template = "sale.ProductCard";
+    static components = { BadgeExtraPrice };
+    static props = {
+        product: ProductProduct,
+        extraPrice: { type: Number, optional: true },
+        onClick: Function,
+        isSelected: { type: Boolean, optional: true },
+    };
+
+    /**
+     * Check whether the provided PTAL should be shown in this card.
+     *
+     * @param {ProductTemplateAttributeLine} ptal The PTAL to check.
+     * @return {Boolean} Whether to show the PTAL.
+     */
+    shouldShowPtal(ptal) {
+        return ptal.hasSelectedCustomPtav || ptal.create_variant === 'no_variant';
```

Comment:
single quoted strings or double quoted strings? the choice is not consistent everywhere :D
(or our guidelines with single for tech strings)

Reply:
Made consistent by following our guidelines. I'll make another pass once all other comments have been addressed.

Note that we're not that consistent with our guidelines in JS files in general (possibly because auto import uses double quotes by default in pycharm).

Reply:
Also because js & website teams did some PRs where we didn't always fight with them over it :D